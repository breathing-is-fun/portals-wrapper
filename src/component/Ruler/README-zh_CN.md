[English](./README.md) | 简体中文

# 简介

- 横向和纵向的刻度尺

# API  

| 名称 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| padding | 比如传了 `15`，那就会被转换成 `padding: 15`；比如传了 `15px`，那就会被转换成 `padding: 15px` | `string | number` | - |

# 示例

``` jsx
    import Ruler from '.'

    render = () => <Ruler>test</Ruler>
```