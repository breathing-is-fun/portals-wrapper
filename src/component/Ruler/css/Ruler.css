.Ruler .wrapper {
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
}

.Ruler .wrapper .cm-horizontal,
.Ruler .wrapper .mm-horizontal {
    position: absolute;
    border-left: 1px solid #555;
    height: 14px;
    width: 9.8%;
}

.Ruler .wrapper .cm-horizontal:after {
    position: absolute;
    bottom: -15px;
    font: 11px/1 sans-serif;
}

.Ruler .wrapper .mm-horizontal {
    height: 5px;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(5) {
    height: 10px;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(1) {
    left: 0%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(1):after {
    content: "";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(2) {
    left: 10%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(2):after {
    content: "1";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(3) {
    left: 20%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(3):after {
    content: "2";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(4) {
    left: 30%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(4):after {
    content: "3";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(5) {
    left: 40%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(5):after {
    content: "4";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(6) {
    left: 50%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(6):after {
    content: "5";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(7) {
    left: 60%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(7):after {
    content: "6";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(8) {
    left: 70%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(8):after {
    content: "7";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(9) {
    left: 80%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(9):after {
    content: "8";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(10) {
    left: 90%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(10):after {
    content: "9";
}

.Ruler .wrapper .cm-horizontal:nth-of-type(11) {
    left: 100%;
}

.Ruler .wrapper .cm-horizontal:nth-of-type(11):after {
    content: "10";
}

.Ruler .wrapper .mm-horizontal:nth-of-type(1) {
    left: 10%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(2) {
    left: 20%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(3) {
    left: 30%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(4) {
    left: 40%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(5) {
    left: 50%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(6) {
    left: 60%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(7) {
    left: 70%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(8) {
    left: 80%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(9) {
    left: 90%;
}

.Ruler .wrapper .mm-horizontal:nth-of-type(10) {
    left: 100%;
}

.Ruler .wrapper .cm-vertical,
.Ruler .wrapper .mm-vertical {
    position: absolute;
    border-top: 1px solid #555;
    width: 14px;
    height: 9.8%;
}

.Ruler .wrapper .cm-vertical:after {
    position: absolute;
    left: 15px;
    font: 11px/1 sans-serif;
}

.Ruler .wrapper .mm-vertical {
    width: 5px;
}

.Ruler .wrapper .mm-vertical:nth-of-type(5) {
    width: 10px;
}

.Ruler .wrapper .cm-vertical:nth-of-type(1) {
    top: 0%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(1):after {
    content: "";
}

.Ruler .wrapper .cm-vertical:nth-of-type(2) {
    top: 10%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(2):after {
    content: "1";
}

.Ruler .wrapper .cm-vertical:nth-of-type(3) {
    top: 20%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(3):after {
    content: "2";
}

.Ruler .wrapper .cm-vertical:nth-of-type(4) {
    top: 30%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(4):after {
    content: "3";
}

.Ruler .wrapper .cm-vertical:nth-of-type(5) {
    top: 40%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(5):after {
    content: "4";
}

.Ruler .wrapper .cm-vertical:nth-of-type(6) {
    top: 50%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(6):after {
    content: "5";
}

.Ruler .wrapper .cm-vertical:nth-of-type(7) {
    top: 60%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(7):after {
    content: "6";
}

.Ruler .wrapper .cm-vertical:nth-of-type(8) {
    top: 70%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(8):after {
    content: "7";
}

.Ruler .wrapper .cm-vertical:nth-of-type(9) {
    top: 80%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(9):after {
    content: "8";
}

.Ruler .wrapper .cm-vertical:nth-of-type(10) {
    top: 90%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(10):after {
    content: "9";
}

.Ruler .wrapper .cm-vertical:nth-of-type(11) {
    top: 100%;
}

.Ruler .wrapper .cm-vertical:nth-of-type(11):after {
    content: "10";
}

.Ruler .wrapper .mm-vertical:nth-of-type(1) {
    top: 10%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(2) {
    top: 20%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(3) {
    top: 30%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(4) {
    top: 40%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(5) {
    top: 50%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(6) {
    top: 60%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(7) {
    top: 70%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(8) {
    top: 80%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(9) {
    top: 90%;
}

.Ruler .wrapper .mm-vertical:nth-of-type(10) {
    top: 100%;
}