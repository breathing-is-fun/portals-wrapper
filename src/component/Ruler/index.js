/*
 * @Author: zy9@github.com/zy410419243
 * @Date: 2018-10-17 19:57:14
 * @Last Modified by: zy9
 * @Last Modified time: 2018-11-21 16:44:40
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './css/Ruler.css';

export default class Ruler extends Component {
	static defaultProps = {
		padding: 35,
	}

	handleScale = type => {
		let cm = [], mm = [];

		// mm
		for(let i = 0; i < 9; i++) {
			const wrapper = (
				<div
					className={ `mm-${ type }` }
					key={ `mm-${ type }-${ i }` }
				/>
			);

			mm.push(wrapper);
		}

		// cm
		for(let i = 0; i < 10; i++) {
			cm.push(
				<div className={ `cm-${ type }` } key={ `cm-${ type }-${ i }` }>
					{ mm }
				</div>
			);
		}

		return cm;
	}

    render = () => {
    	const { children, padding } = this.props;

    	return (
    		<div className='Ruler'>
    			<div className='wrapper'>
    				{ this.handleScale('horizontal') }
    			</div>

    			<div className='wrapper'>
    				{ this.handleScale('vertical') }
    			</div>

    			<div style={{ padding }}>
    				{ children }
    			</div>
    		</div>
    	);
    }
}

Ruler.propTypes = {
	padding: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
	]),
};