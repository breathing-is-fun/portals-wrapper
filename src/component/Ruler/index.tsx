import React, { Component } from 'react';
import './css/Ruler.css';

export interface RulerProps {
  padding: number | string;
}

export default class Ruler extends Component<RulerProps> {
  static defaultProps = {
    padding: 35,
  };

  handleScale = (type: string) => {
    let cm = [],
      mm = [];

    // mm
    for (let i = 0; i < 9; i++) {
      const wrapper = <div className={`mm-${type}`} key={`mm-${type}-${i}`} />;

      mm.push(wrapper);
    }

    // cm
    for (let i = 0; i < 10; i++) {
      cm.push(
        <div className={`cm-${type}`} key={`cm-${type}-${i}`}>
          {mm}
        </div>,
      );
    }

    return cm;
  };

  render = () => {
    const { children, padding } = this.props;

    return (
      <div className="Ruler">
        <div className="wrapper">{this.handleScale('horizontal')}</div>

        <div className="wrapper">{this.handleScale('vertical')}</div>

        <div style={{ padding }}>{children}</div>
      </div>
    );
  };
}
