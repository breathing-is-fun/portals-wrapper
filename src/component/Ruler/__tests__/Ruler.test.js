/*
 * @Author: zy9@github.com/zy410419243
 * @Date: 2018-10-24 17:14:50
 * @Last Modified by: zy9
 * @Last Modified time: 2018-10-24 18:53:36
 */
import React from 'react';
import { render, mount, shallow } from 'enzyme';
import Ruler from '..';

describe('Navigation', () => {
	it('render correctly', () => {
		const wrapper = mount(<Ruler>test</Ruler>);

		expect(wrapper).toMatchSnapshot();
	});
});