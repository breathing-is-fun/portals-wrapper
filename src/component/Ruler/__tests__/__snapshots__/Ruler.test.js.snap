// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation render correctly 1`] = `
<Ruler>
  <div
    className="Ruler"
  >
    <div
      className="wrapper"
    >
      <div
        className="cm-horizontal"
        key="cm-horizontal-0"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-1"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-2"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-3"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-4"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-5"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-6"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-7"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-8"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
      <div
        className="cm-horizontal"
        key="cm-horizontal-9"
      >
        <div
          className="mm-horizontal"
          key="mm-horizontal-0"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-1"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-2"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-3"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-4"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-5"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-6"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-7"
        />
        <div
          className="mm-horizontal"
          key="mm-horizontal-8"
        />
      </div>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="cm-vertical"
        key="cm-vertical-0"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-1"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-2"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-3"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-4"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-5"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-6"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-7"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-8"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
      <div
        className="cm-vertical"
        key="cm-vertical-9"
      >
        <div
          className="mm-vertical"
          key="mm-vertical-0"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-1"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-2"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-3"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-4"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-5"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-6"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-7"
        />
        <div
          className="mm-vertical"
          key="mm-vertical-8"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "padding": 20,
        }
      }
    >
      test
    </div>
  </div>
</Ruler>
`;
